### DISCUSSION TITLE
Add query parameter to configure default graph as union of all graphs

### DISCUSSION THREAD

**Discussion by rat10** [oxigraph/oxigraph#1133] [Ideas]
Hi there, I'm using Oxigraph as part of [Zazuko's "SPARQL Notebook" VScode extension](https://github.com/zazuko/vscode-sparql-notebook). I'd like simple queries like "SELECT * WHERE { ?s ?p ?o }" to run over _all_ graphs, not only the default graph (which is to spec, but can get quite uncomfortable quite quickly). Ideally (in my case), but not essential, that union graph would be a _union_, not a _merge_ , of all named graphs _and_ the default graph.

I see that some option to this effect has already been added  in different places like the CLI and JS interfaces (although I'm not clear about the details, see above). Some graph databases [provide](https://github.com/w3c/sparql-dev/issues/43#issuecomment-694144595) [keywords](https://www.poolparty.biz/blogposts/rdf-graph-databases-and-their-default-graph/) that can be used in a FROM clause (although different ones, as this is [not yet standardized](https://github.com/w3c/sparql-dev/issues/43)). Is it possible to add such a query parameter to Oxigraph, or is it already available and I'm just not aware?

**Comment by Tpt** [comment]
Thank you for raising this. There is indeed no way to query the union graph using only SPARQL syntax in Oxigraph at the moment. However, if you add the `union-default-graph` query parameter to the endpoint url it will work eg. `http://localhost:7878/query?union-default-graph`. I don't know if setting query parameters is supported by the SPARQL Notebook VScode extension.

**Reply by rat10** [reply]
Hi Thomas, thank you for your quick response! I'll check with Zazuko if the connection to the internal Oxigraph DB can be configured accordingly and will update this reply in case I find a solution.

**Comment by rat10** [comment]
Zazuko has now added a parameter, to that effect, e.g.:

```
# [endpoint=./rdf/a.trig]
# [use_default_graph_as_union=true]


SELECT * WHERE {
    ?s ?p ?o
}
```

See the [comment](https://github.com/zazuko/vscode-sparql-notebook/issues/90#issuecomment-2662545711) on their GitHub page.

**Comment by rat10** [comment]
My next wish ;-)  would be an IRI or URN to refer to the standard default graph, something like `oxi:default` or `<urn:x-sparql:default>`...
[EDIT] After more experiments with other graph stores I'm sorry to have to extend my wish list with more proper references:
- default graph
- union of all named graphs (excluding the default graph)
- union of all graphs (including the default graph)
